src/toolkit.cpp

```diff
@@ -695,6 +695,9 @@ bool Toolkit::LoadData(const std::string &data)
         }
     }
 
+    // generate the measure indices
+    m_doc.GenerateMeasureIndices();
```

Comment:
You are right, even though this is a bit more involving because it would require looking for previously generated number. I am fine to leave it like this for now, maybe with just a comment in the code.